CrimeMonitor/CrimeMonitor.cs

```diff
@@ -0,0 +1,849 @@
+                }
+
+                if (record != null)
+                {
+                    // Get all bounty claims, excluding the discrepancy report
+                    List<FactionReport> reports = record.factionReports
+                        .Where(r => r.bounty && r.crimeDef == Crime.None).ToList();
+                    if (reports?.Any() ?? false)
+                    {
+                        long total = reports.Sum(r => r.amount);
+
+                        // Check for discrepancy in logged bounty claims
+                        if (total < amount)
+                        {
+                            // Adjust the discrepancy report & remove when zeroed out
+                            FactionReport report = record.factionReports
+                                .FirstOrDefault(r => r.crimeDef == Crime.Claim);
+                            if (report != null)
+                            {
+                                report.amount -= Math.Min(amount - total, report.amount);
+                                if (report.amount == 0) { reports.Add(report); }
+                            }
+                        }
+                        // Remove associated bounty claims
+                        record.factionReports = record.factionReports.Except(reports).ToList();
+                    }
+                    // Adjust the total claims
+                    record.claims -= Math.Min(amount, record.claims);
+
+                    RemoveRecord(record);
+                    update = true;
+                }
+            }
+            return update;
+        }
+
+        private void handleBountyIncurredEvent(BountyIncurredEvent @event)
+        {
+            if (@event.timestamp > updateDat)
+            {
+                updateDat = @event.timestamp;
+                _handleBountyIncurredEvent(@event);
+                writeRecord();
+            }
+        }
+
+        private void _handleBountyIncurredEvent(BountyIncurredEvent @event)
+        {
+            int shipId = EDDI.Instance?.CurrentShip?.LocalId ?? 0;
+            Crime crime = Crime.FromEDName(@event.crimetype);
+            string currentSystem = EDDI.Instance?.CurrentStarSystem?.name;
+            FactionReport report = new FactionReport(@event.timestamp, true, shipId, crime, currentSystem, @event.bounty)
+            {
+                station = EDDI.Instance?.CurrentStation?.name,
+                body = EDDI.Instance?.CurrentStellarBody?.name,
+                victim = @event.victim
+            };
+
+            FactionRecord record = GetRecordWithFaction(@event.faction);
+            if (record == null)
+            {
+                record = AddRecord(@event.faction);
+            }
+            record.factionReports.Add(report);
+            record.bounties += report.amount;
+        }
+
+        private void handleBountyPaidEvent(BountyPaidEvent @event)
+        {
+            if (@event.timestamp > updateDat)
```

Comment:
See above. Move this if statement to the PreHandle method (rather than repeating it for each individual event)?

Reply:
Concur. Done.